tests/test_dataclasses_support.py

```diff
@@ -0,0 +1,39 @@
+import sys
+
+import pytest
+
+import andi
+
+
+@pytest.mark.skipif(sys.version_info != (3, 6),
+                    reason="Testing dataclasses backport for Python 3.6")
+def test_dataclasses_py36():
+    from py36 import ADC, BDC
+    assert andi.inspect(ADC.__init__) == {'b': [BDC]}
+    assert andi.inspect(BDC.__init__) == {'a': [ADC]}
+
+
+@pytest.mark.skipif(sys.version_info < (3, 7, 6),
+                    reason="Dataclasses require Python 3.7.6 or higher")
+def test_dataclasses_py37():
+    from py37 import ADC, BDC
+    assert andi.inspect(ADC.__init__) == {'b': [BDC]}
+    assert andi.inspect(BDC.__init__) == {'a': [ADC]}
+
+
+@pytest.mark.skipif(sys.version_info < (3, 7, 6),
+                    reason="Dataclasses require Python 3.7.6 or higher")
+def test_dataclasses_py37_no_pep_563():
+    from py37_no_pep_563 import ADC, BDC
+    assert andi.inspect(ADC.__init__) == {'b': [BDC]}
+    assert andi.inspect(BDC.__init__) == {'a': [ADC]}
```

Comment:
Do regular type references work with dataclasses in Python 3.7, i.e. no strings, no future imports? If so, what do you think about adding a test for it?

Reply:
Yes, they work. It was already tested, but I rearranged the tests  here: https://github.com/scrapinghub/andi/pull/3/commits/f773dc5ba86d8cc84cfe596b755ad74387398582 to make it more clear.

Reply:
Ok, wait. It seems it fails... Let me have a second look.

Reply:
Ok, so finally this is the status of dataclasses support:
- Regular dataclasses are supported. 
- Any kind of forward references (using or not the future import) are only supported since 3.7.6 and 3.8.1

In this commit the above is tested: https://github.com/scrapinghub/andi/pull/3/commits/8b3e2542d83b1f6a1ab88be1940c0848f094d4a9